### ISSUE TITLE
OOM exception happening after upgrading to v0.10.0

### ISSUE THREAD

**Issue by R4md4c** [coil-kt/coil#368]
**Describe the bug**
Upgrading to 0.10.0 causes a crash when trying to load a screen sized local asset as background, even though the exact same request works for v0.9.5

**Expected behavior**
It should load with no crash.

**To Reproduce**

I'm using the ImageLoader API 
```
val drawable = imageLoader.get(
          GetRequestBuilder(context)
                .data(R.drawable.bg_slider)
                .build()
     )
```

and here is how I built it 

```
    override suspend fun init() {
        Coil.setImageLoader(this)
    }

    override fun newImageLoader(): ImageLoader {
        return ImageLoaderBuilder(context)
            .okHttpClient {
                okHttpClient.get().newBuilder()
                    .cache(CoilUtils.createDefaultCache(context))
                    .build()
            }
            .build()
    }
```

**Logs/Screenshots**
```
java.lang.OutOfMemoryError: Failed to allocate a 327680012 byte allocation with 4194304 free bytes and 95MB until OOM
        at java.io.BufferedInputStream.fillbuf(BufferedInputStream.java:163)
        at java.io.BufferedInputStream.skip(BufferedInputStream.java:378)
        at java.io.DataInputStream.skipBytes(DataInputStream.java:204)
        at androidx.exifinterface.media.ExifInterface$ByteOrderedDataInputStream.skipBytes(ExifInterface.java:7654)
        at androidx.exifinterface.media.ExifInterface.getPngAttributes(ExifInterface.java:6077)
        at androidx.exifinterface.media.ExifInterface.loadAttributes(ExifInterface.java:4568)
        at androidx.exifinterface.media.ExifInterface.<init>(ExifInterface.java:4032)
        at androidx.exifinterface.media.ExifInterface.<init>(ExifInterface.java:3983)
        at coil.decode.BitmapFactoryDecoder.decode(BitmapFactoryDecoder.kt:53)
        at coil.RealImageLoader$loadData$2.invokeSuspend(RealImageLoader.kt:354)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

**Version**
v0.10.0
Emulator: Galaxy_Nexus_API_23

**Comment by colinrtwhite** [comment]
If it's possible can you post the image that's crashing? Seems like it's either an issue in the EXIF parser or the image's EXIF data.

**Comment by R4md4c** [comment]
@colinrtwhite just in case you haven't seen my email, I've sent the image under `Github Issue #368 Request Image` subject.

Thank you in advance.

**Comment by colinrtwhite** [comment]
@R4md4c Thanks for the report + sample image. The fix is currently in the [snapshots](https://coil-kt.github.io/coil/faq/#how-do-i-get-development-snapshots) and will be available in `0.10.1` either today or tomorrow.
